### PR TITLE
Chore(Test): Addition of E2E test for Azure Experiments

### PR DESCRIPTION
**Pull request by avaakash**
## Proposed changes

This PR adds the E2E test for the Azure Instance Stop Experiment and Azure Disk Loss Experiment
Link to Sample Run: https://github.com/avaakash/litmus-e2e/runs/4758029750
Partially completes Issue #308 

## How has this been tested:
- [ ] Test environment
    * [ ] GKE
    * [ ] AWS
    * [ ] OpenShift
    * [ ] KinD/k3s/k3d cluster
    * [x] Others
 - [ ] Tested with production environment ( For ChaosCenter Tests )
 - [ ] Have not tested

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus-e2e/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [ ] I have added necessary documentation (if appropriate)
- [x] I have added Screenshots or Terminal snippets in the PR comment to validate the successful execution of changes

**Documentation**
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] data-cy=* parameters have been added in codebase ( ChaosCenter Tests )
- [ ] data-cy=* parameters need to be added in codebase ( ChaosCenter Tests )
- [x] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. uncaught exceptions from 3rd party library (if any)

**Issue**
- [ ] Tasks in issue are checked off

## Dependency
- Please add the links to the dependent PR that need to be merged before this (if any).

## Special notes for your reviewer:
Following secrets need to be added for the e2e tests
1. AZURE_CREDENTIALS - Complete auth file content for Azure CLI login
2. AZURE_AUTH_FILE - Complete auth file content with escaped quotes character
3. AZURE_RESOURCE_GROUP - resource group name for azure
4. DISK_ONE - Disk name
5. DISK_TWO - Disk name
6. INSTANCE_ONE - Instance name
7. INSTANCE_TWO - Instance name

### CODE REVIEW

File: .github/workflows/azure-experiment-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Create a Azure VM Instance with target Disk Volumes
+        if: always()
+        run: |
+          az group create -n ${{ secrets.AZURE_RESOURCE_GROUP }}_disk_${{ github.run_number }} -l eastus && \
+          az vm create -g ${{ secrets.AZURE_RESOURCE_GROUP }}_disk_${{ github.run_number }} -n ${{ secrets.INSTANCE_ONE }} --image UbuntuLTS --generate-ssh-keys && \
+          az disk create -n ${{ secrets.DISK_ONE }} -g ${{ secrets.AZURE_RESOURCE_GROUP }}_disk_${{ github.run_number }} --size-gb=4 && \
+          az disk create -n ${{ secrets.DISK_TWO }} -g ${{ secrets.AZURE_RESOURCE_GROUP }}_disk_${{ github.run_number }} --size-gb=4 && \
```
Comment (neelanjan00) [inline .github/workflows/azure-experiment-pipeline.yml:154]: Can we not create secrets for the names of the disk and mention them directly?

File: .github/workflows/nightly-azure-experiment-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Create target Azure VM Instances
+        if: always()
+        run: |
+          az group create -n ${{ secrets.AZURE_RESOURCE_GROUP }}_instance_${{ github.run_number }} -l eastus && \
+          az vm create -g ${{ secrets.AZURE_RESOURCE_GROUP }}_instance_${{ github.run_number }} -n ${{ secrets.INSTANCE_ONE }} --image UbuntuLTS --generate-ssh-keys && \
+          az vm create -g ${{ secrets.AZURE_RESOURCE_GROUP }}_instance_${{ github.run_number }} -n ${{ secrets.INSTANCE_TWO }} --image UbuntuLTS --generate-ssh-keys
```
Comment (neelanjan00) [inline .github/workflows/nightly-azure-experiment-pipeline.yml:48]: Can we not create secrets for the instance names and mention them directly?

File: .github/workflows/nightly-azure-experiment-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Create a Azure VM Instance with target Disk Volumes
+        if: always()
+        run: |
+          az group create -n ${{ secrets.AZURE_RESOURCE_GROUP }}_disk__${{ github.run_number }} -l eastus && \
+          az vm create -g ${{ secrets.AZURE_RESOURCE_GROUP }}_disk__${{ github.run_number }} -n ${{ secrets.INSTANCE_ONE }} --image UbuntuLTS --generate-ssh-keys && \
+          az disk create -n ${{ secrets.DISK_ONE }} -g ${{ secrets.AZURE_RESOURCE_GROUP }}_disk__${{ github.run_number }} --size-gb=4 && \
+          az disk create -n ${{ secrets.DISK_TWO }} -g ${{ secrets.AZURE_RESOURCE_GROUP }}_disk__${{ github.run_number }} --size-gb=4 && \
```
Comment (neelanjan00) [inline .github/workflows/nightly-azure-experiment-pipeline.yml:134]: Can we not create secrets for the disk names and mention them directly?

File: .master-plan.yml [hunk 4]
```diff
@@ -230,6 +230,28 @@ spec:
       git/location: ""
       test/status: "Not Done"
 
+
+  # -------------------------------------------------
+  # Kubernetes Azure chaos experiment BDD Tests
+  # -------------------------------------------------   
+
+  - tcid: TCID-EC2-AZURE-INFRA-INSTANCE-STOP
+    name: "TCID-EC2-AZURE-INFRA-INSTANCE-STO"
```
Comment (neelanjan00) [inline .master-plan.yml:239]:
```suggestion
    name: "TCID-EC2-AZURE-INFRA-INSTANCE-STOP"
```

File: .github/workflows/azure-experiment-pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+          curl -sfL https://get.k3s.io | sh -s - --docker --write-kubeconfig-mode 664
+          kubectl wait node --all --for condition=ready --timeout=90s
+          mkdir -p $HOME/.kube && cat /etc/rancher/k3s/k3s.yaml > $HOME/.kube/config          
+          kubectl get nodes
+          
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.14'
```
Comment (uditgaurav) [inline .github/workflows/azure-experiment-pipeline.yml:131]: this is quite older can we use 1.16>

File: .github/workflows/nightly-azure-experiment-pipeline.yml [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+          curl -sfL https://get.k3s.io | sh -s - --docker --write-kubeconfig-mode 664
+          kubectl wait node --all --for condition=ready --timeout=90s
+          mkdir -p $HOME/.kube && cat /etc/rancher/k3s/k3s.yaml > $HOME/.kube/config          
+          kubectl get nodes
+          
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.14'
```
Comment (uditgaurav) [inline .github/workflows/nightly-azure-experiment-pipeline.yml:111]: pls check this at other places too

**Review by uditgaurav — APPROVED** [review]

**Review by neelanjan00 — APPROVED** [review]

**Pull request merged**